File: haskell/fizzbuzz.hs
{-
	FizzBuzz

	To compile:
	$ ghc -o bin/fizzbuzz fizzbuzz.hs

	To run:
	./bin/fizzbuzz
-}

fizz :: Int -> String
fizz n | n `mod` 15 == 0  = "FizzBuzz"
       | n `mod` 3  == 0  = "Fizz"
       | n `mod` 5  == 0  = "Buzz"
       | otherwise        = show n

main :: IO()
main = mapM_ putStrLn $ map fizz [1..100]

File: c/fizzbuzz.c
/**
 * FizzBuzz
 *
 * Compile with:
 * $ c99 fizzbuzz.c -o bin/fizzbuzz
 * 
 * Run with:
 * $ ./bin/fizzbuzz
 */

#include <stdio.h>

int main()
{
	int i;

	for( i = 1; i <= 100; i++ ) {
		if( i % 15 == 0 ) {
			printf( "FizzBuzz\n" );
		} else if( i % 3 == 0 ) {
			printf( "Fizz\n" );
		} else if( i % 5 == 0 ) {
			printf( "Buzz\n" );
		} else {
			printf( "%d\n", i );
		}
	}

	return 0;
}
